### PR TITLE
working Cloudcare for child modules

### PR DESCRIPTION
**Pull request by snopoke**
@czue 

cc @benrudolph @wpride

### CODE REVIEW

File: corehq/apps/cloudcare/touchforms_api.py [hunk 1]
```diff
@@ -82,6 +50,63 @@ def filter_cases(self, xpath, additional_filters=None, auth=None, extra_instance
...
+        if self.case_id:
+            if self.delegation:
+                session_data["delegation_id"] = self.case_id
+                session_data["case_id"] = self._case_parent_id
+            else:
+                session_data[self.case_session_variable_name] = self.case_id
+        return session_data
+
+    @property
+    def case_session_variable_name(self):
```
Comment (snopoke) [inline corehq/apps/cloudcare/touchforms_api.py]: This is the key change

File: corehq/apps/app_manager/suite_xml/sections/entries.py [hunk 2]
```diff
@@ -18,7 +18,11 @@
 from corehq.apps.app_manager.suite_xml.xml_models import *
 
 
-FormDatumMeta = namedtuple('FormDatumMeta', 'datum case_type requires_selection action')
+class FormDatumMeta(namedtuple('FormDatumMeta', 'datum case_type requires_selection action')):
+    def __repr__(self):
+        return 'FormDataumMeta(datum=<SessionDatum(id={})>, case_type={}, requires_selection={}, action={})'.format(
```
Comment (dimagimon) [inline corehq/apps/app_manager/suite_xml/sections/entries.py]: E501 line too long (116 > 115 characters)

File: corehq/apps/smsforms/app.py [hunk 3]
```diff
@@ -41,8 +41,8 @@ def start_session(domain, contact, app, module, form, case_id=None, yield_respon
     """
     # NOTE: this call assumes that "contact" will expose three
     # properties: .raw_username, .get_id, and .get_language_code
-    session_data = get_session_data(domain, contact, case_id, device_id=COMMCONNECT_DEVICE_ID)
-    
+    session_data = CaseSessionDataHelper(domain, contact, case_id, app, form).get_session_data(COMMCONNECT_DEVICE_ID)
```
Comment (dimagimon) [inline corehq/apps/smsforms/app.py]: E501 line too long (117 > 115 characters)

**Comment by czue** [comment]
classic sneak attack! :guardsman:

File: corehq/apps/smsforms/app.py [hunk 4]
```diff
@@ -41,7 +41,7 @@ def start_session(domain, contact, app, module, form, case_id=None, yield_respon
     """
     # NOTE: this call assumes that "contact" will expose three
     # properties: .raw_username, .get_id, and .get_language_code
-    session_data = get_session_data(domain, contact, case_id, device_id=COMMCONNECT_DEVICE_ID)
+    session_data = SessionDataHelper(domain, contact, case_id).get_session_data(COMMCONNECT_DEVICE_ID)
```
Comment (benrudolph) [inline corehq/apps/smsforms/app.py]: :joy:

**Comment by benrudolph** [comment]
solid and nice cleanup :bath:

**Pull request merged**